woocommerce/woocommerce
# Add basic e2e setup and tests for the Assembler Hub

### Changes proposed in this Pull Request:

This adds some setup/configuration code that will allow us to test Assembler:

* Allow switching to a block theme during testing, I used `wp-env` + `wp-cli` in shell for this.
* Allow setting wp options during testing. This allows me to disable the welcome tour for the tests I wrote here because in CI environment it blocks clicks of some elements.
* Allow configuring enabled/disabled feature flags, this builds on a recent plugin that was introduced to set a feature flag and this allows you to enable/disable feature flags before running a test.

And this adds a couple of basic smoke tests:

1. Assembler hub loads. I haven't asserted the BlockPreview yet, its quite slow to load, but this will confirm at least that the assembler has not crashed

2. Confirm we can navigate to header customization and see some block patterns populated.





### How to test the changes in this Pull Request:



See that CI passes, but (optional) you can also run them locally and inspect the whole test run:

1. set up playwright env as per the docs: 

```markdown
**Running tests for the first time:**

-   `nvm use` (uses the default node version you have set in NVM)
-   `pnpm install` (installs dependencies)
-   `pnpm run build --filter=woocommerce` (builds WooCommerce locally)
-   `cd plugins/woocommerce` (changes into the WooCommerce plugin folder)
-   `npx playwright install` (installs the latest Playwright version)
-   `pnpm env:start` (starts the local environment)
```
2. You'll need to activate the helper plugin, I think the script that activates it is not triggered when running individual tests right now. I just logged in to the wp-env instance after `env:start` and activated it manually. It is called "Test Helper APIs"
3. Run this suite in ui mode: `pnpm test:e2e-pw --ui ./tests/e2e-pw/tests/customize-store/assembler-hub.spec.js`




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add some basic E2E tests for Assembler Hub

#### Comment 

</details>
